framework/quota/src/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -358,10 +358,11 @@ public QuotaUsageVO updateQuotaDiskUsage(UsageVO usageRecord, final BigDecimal a
         BigDecimal rawusage;
         // get service offering details
         ServiceOfferingVO serviceoffering = _serviceOfferingDao.findServiceOffering(usageRecord.getVmInstanceId(), usageRecord.getOfferingId());
+        if (serviceoffering == null) return quotalist;
         rawusage = new BigDecimal(usageRecord.getRawUsage());
 
         QuotaTariffVO tariff = _quotaTariffDao.findTariffPlanByUsageType(QuotaTypes.CPU_NUMBER, usageRecord.getEndDate());
-        if (tariff != null && tariff.getCurrencyValue().compareTo(BigDecimal.ZERO) != 0) {
+        if (tariff != null && tariff.getCurrencyValue().compareTo(BigDecimal.ZERO) != 0 && serviceoffering.getCpu() != null) {
```

Comment:
Hello @agneya2001 ,

In the refactor you plan to make, you could also turn this piece of code into a method, for reusability:
```Java
tariff != null && tariff.getCurrencyValue().compareTo(BigDecimal.ZERO)
```